packages/node-core/src/indexer/poi/PoiBlock.ts

```diff
@@ -12,57 +12,84 @@ import {
   base58Decode,
   isBase64,
   base64Decode,
+  u8aToBuffer,
 } from '@subql/utils';
 import {ProofOfIndex} from '../entities/Poi.entity';
+export const DEFAULT_BLOCK_HASH = hexToU8a('0x00');
+export const NULL_OPERATION_HASH = hexToU8a('0x00');
 
 const poiBlockHash = (
   id: number,
-  chainBlockHash: string | Uint8Array,
-  operationHashRoot: Uint8Array,
-  parentHash: Uint8Array,
-  projectId: string
+  chainBlockHash: string | Uint8Array | null,
+  operationHashRoot: Uint8Array | null,
+  projectId: string,
+  parentHash: Uint8Array
 ): Uint8Array => {
-  if (!id || !chainBlockHash || !operationHashRoot || !projectId) {
+  if (!id || !projectId || chainBlockHash === undefined || operationHashRoot === undefined) {
     throw new Error('Poof of index: can not generate block hash');
   }
-  return blake2AsU8a(u8aConcat(numberToU8a(id), chainBlockHash, operationHashRoot, Buffer.from(projectId), parentHash));
+  return blake2AsU8a(
+    u8aConcat(
+      numberToU8a(id),
+      chainBlockHash ?? u8aToBuffer(DEFAULT_BLOCK_HASH),
+      operationHashRoot ?? u8aToBuffer(NULL_OPERATION_HASH),
+      Buffer.from(projectId),
+      parentHash
+    )
+  );
 };
 
 export class PoiBlock implements ProofOfIndex {
   readonly id: number;
-  readonly chainBlockHash: Uint8Array;
-  readonly hash: Uint8Array;
-  readonly parentHash: Uint8Array;
-  readonly operationHashRoot: Uint8Array;
+  readonly chainBlockHash: Uint8Array | null;
+  readonly operationHashRoot: Uint8Array | null;
   readonly projectId: string;
+  readonly parentHash: Uint8Array | undefined;
+  readonly hash: Uint8Array | undefined;
 
   constructor(
     id: number,
-    chainBlockHash: Uint8Array,
-    hash: Uint8Array,
-    parentHash: Uint8Array,
-    operationHashRoot: Uint8Array,
-    projectId: string
+    chainBlockHash: Uint8Array | null,
```

Comment:
Could we change this to a string? Currently we need to ensure we can convert any block hash format to Uint8Array so when we add support for a new chain if they provide a different encoding we need to make changes here. 

It would help remove the big if/else block in create

Reply:
This will require convert any exiting chainBlockHash in db convert from `bytea` to `text`. It will increase complexity for migration, we can leave this now